src/screencast/wlr_screencast.c

```diff
@@ -295,6 +295,213 @@ struct xdpw_wlr_output *xdpw_wlr_output_first(struct wl_list *output_list) {
 	return NULL;
 }
 
+static bool exec_chooser_simple(char *cmd, char *name, unsigned long name_maxlength) {
+	int p1[2]; //p -> c
+	int p2[2]; //c -> p
+
+	if (pipe(p1) == -1) {
+		logprint(ERROR,"Failed to open pipe");
+		return false;
+	}
+	if (pipe(p2) == -1) {
+		logprint(ERROR,"Failed to open pipe");
+		return false;
+	}
+
+	pid_t pid = fork();
+
+	if (pid < 0) {
+		perror("fork");
+		return false;
+	} else if (pid == 0) {
+		close(p1[1]);
+		close(p2[0]);
+
+		dup2(p1[0], STDIN_FILENO);
+		dup2(p2[1], STDOUT_FILENO);
+		close(p1[0]);
+		close(p2[1]);
+
+		int err;
+		err = execl("/bin/sh", "/bin/sh", "-c", cmd, (void *)NULL);
+
+		if (err == -1) {
+			perror("execl");
+			logprint(WARN,"Failed to execute %s",cmd);
+			return false;
+		}
+		exit(127);
+	}
+
+	close(p1[0]);
+	close(p2[1]);
+	close(p1[1]);
+	if (read(p2[0],name,name_maxlength*sizeof(char)) == -1) {
+		return false;
+	}
+
+	name[name_maxlength -1] = '\0';
+	char *p = strchr(name, '\n');
```

Comment:
I think this is the issue here. I'm guessing this is how we're attempting to handle output names shorter than `name_maxlength`. A simpler approach might be to initialize `name` to all `\0` before reading into it. That way, wherever it ends, it is null terminated. Then we could still replace any instances of control characters like newlines.

The maxlength approach in general is a lot of code that adds complexity. Maybe we can just assume that output names will never be longer than some fixed size, like 1024 bytes?

@emersion, any suggestions? I'm not sure what assumptions we can make about output names in wlroots.

Reply:
Does this recent commit solves it?

About the maxlenght approach, I don't like it, but I don't know a more complete way, if wlroots has no max output namelength.

Reply:
Well, we could try dynamically allocating a string in the read loop, but that's frustrating for such a small thing.

Reply:
I can confirm that this last commit fixes the errors I'm seeing with garbage on the output both for hitting escape and for slurp. I'll mark this resolved and we can revisit string handling when we address config files and user configurable choosers.